lib/galaxy/managers/folders.py

```diff
@@ -309,3 +312,257 @@ def cut_and_decode(self, trans, encoded_folder_id):
         :rtype:    int
         """
         return self.decode_folder_id(trans, self.cut_the_prefix(encoded_folder_id))
+
+
+class FoldersManager:
```

Comment:
What is the difference between the FolderManager and FoldersManager? Should these be the same class?

Reply:
The `s` character... just kidding 😆 

The idea is that `FoldersManager` is an abstraction layer that directly contains the logic used in the `FoldersController` API controller and uses `FolderManager` and `RoleManager` to implement it. This shared (with the FastAPI controller) manager will also use and return the future pydantic models.
But yeah... I agree the name is not the best... 😅 

Do you think these two should go together?

Reply:
@jmchilton 
Do you prefer [this approach](https://github.com/galaxyproject/galaxy/pull/11686/files) to deal with the class encapsulating the controller logic instead of calling it a `Manager`? Or am I going completely offroad here? 😆 

Appreciated any advice on how to proceed with this :)

Reply:
I have renamed the `FoldersManager` to `FoldersService` to better differentiate the class encapsulating the API actions logic from the regular Managers used to deal with the database models.
I have also used this naming in #11712 to unify it a bit. Again, the idea is to have this layer that directly speaks "pydantic" and uses the appropriate managers under the hood to expose the API actions that can be then called from the API controllers (either legacy or FastAPI) in a one-liner to reuse as much logic as possible and keed the API controllers thin.

But I am really open to better naming ideas or changes in this approach :)